src/component/mxgraph/shape/gateway-shapes.ts

```diff
@@ -38,7 +38,21 @@ abstract class GatewayShape extends mxRhombus {
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   protected paintInnerShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
-    // do nothing by default
+    c.setFillColor(this.stroke);
+    c.setStrokeWidth(0);
+  }
+  protected getScaledGeometry(x: number, y: number, w: number, h: number): { xS: number; yS: number; wS: number; hS: number } {
```

Comment:
:question: good to share code! :smile: 
I have noticed that we didn't communicate well about the icon scaling and position of elements in the task (what I did) and gateway (what you did) shapes and we now have 2 ways of doing things :hushed: 

I have introduced a `MxScaleFactorCanvas` class in charge of hidding the scaling
```typescript
// vanilla canvas calls when a scale factor must be applied to positions
const scaleFactor = 0.26;
c.moveTo(8 * scaleFactor, 39 * scaleFactor);
c.lineTo(12 * scaleFactor, 25 * scaleFactor);

// MxScaleFactorCanvas
const canvas = new MxScaleFactorCanvas(c, 0.26);
canvas.moveTo(8, 39);
canvas.lineTo(12, 25);
```
This makes the code clearer than in the current gateway implementation. However, it only scales the x and y direction with the same value.

In the gateway side, you introduce a way to manage the translation required to position the icon. In the task shape, this is done globally by a translation of the current cursor which has a side effect: after icon drawing, the orgin coordinates have changed, so subsequent drawing won't be done at the right place, except if we remember to reset the origin coordinates at the end of the task icon drawing! For sure, this will create issue in the future when we will draw the task markers.

So to me, we have to merge the good part of the 2 implementations in a new single one, especially if we want to benefits of it for event icons.

I suggest that we keep the gateway icon implementation as it is now in this PR but we manage a refactoring very soon (let's say next week) with @csouchet as well. If you agree, I can create an issue to track this.

Reply:
as discussed with @aibcmars, this will be managed in #226

Reply:
yes, done in 226 so resolving this conversation

Reply:
not fully implemented in #226, so unresolving